### DISCUSSION TITLE
Securing many to many model to read partial data from foriegn key

### DISCUSSION THREAD

**Discussion by heauton** [supabase/supabase#1762] [Questions] [answered]
I am following https://github.com/supabase/supabase/discussions/710 in a combination with security examples outlined in https://supabase.io/docs/guides/auth#policies-with-security-definer-functions. 

But once I enable security rules for `users` table:

```sql
create policy "Can read my data" on public.users for select using (auth.uid() = id);
```

I can't get other team members but the current user with this query:

```ts
  const {data: teams, error} = await supabase
    .from('teams')
    .select(`*, users!team_members(*)`)
```

Ideally, I would want to:

1. Get team data as well as associated team members
2. Let team members read partial data from `users` like avatar, name, etc. without exposing entire columns

I tried creating user profiles view:

```sql
create view public.user_profiles
  as select name, avatar_url from public.users;
``` 

But I am not sure how to use this when querying or when creating policies. Or there could be some solutions other than using views but I am not familiar with SQL. Could anyone help me point in the right direction?

**Comment by steve-chavez** [comment, marked as answer]
Hey @heauton,

Really sorry for the late reply here, this fell off my radar.

Using a view is fine, you can use it like any other table with [supabase.select()](https://supabase.io/docs/reference/javascript/select). For using RLS on views, there's a gotcha that's covered on this [previous discussion](https://github.com/supabase/supabase/discussions/1501).
